runtime/lua/vim/lsp.lua

```diff
@@ -1048,14 +1045,16 @@ function lsp.start_client(config)
   ---@private
   --- Sends a notification to an LSP server.
   ---
-  ---@param method (string) LSP method name.
-  ---@param params (optional, table) LSP request params.
-  ---@param bufnr (number) Buffer handle, or 0 for current.
+  ---@param method string LSP method name.
+  ---@param params table|nil LSP request params.
   ---@returns {status} (bool) true if the notification was successful.
   ---If it is false, then it will always be false
   ---(the client has shutdown).
-  function client.notify(...)
-    return rpc.notify(...)
+  function client.notify(method, params)
+    if method ~= 'textDocument/didChange' then
+      changetracking.flush(client)
+    end
```

Comment:
This feels a bit hacky and so far things worked without it, but I think we ought to do this to ensure the server state is up2date before sending other notifications. 

(Side effect of this change is that we don't need `vim.wait` calls in the test, because `client.notify('finish')` will flush the `didChange` first)

Reply:
I was looking at this hah.

Reply:
I am ok with this, plus there aren't that many notifications. I'm not sure what the correct thing to do (in an ideal world) would even be... interpolate the notifications with the textdocument/didChange notifications?